src/context.rs

```diff
@@ -291,9 +291,6 @@ impl Context {
     /// Add a record to the reverse lookup cache
     #[cfg(feature = "local-dns-relay")]
     pub fn add_to_reverse_lookup_cache(&self, addr: &IpAddr, forward: bool) {
-        if self.check_ip_in_proxy_list(addr) == forward {
-            return;
-        }
```

Comment:
This optimization should probably be fine?

Reply:
It's possible that the domain name is marked proxied, but IP is not in the proxy list. Typically in GFW List mode.

Reply:
Hmm maybe we should instead check whether the outcome is consistent with the cache and add/remove from cache accordingly. (I want to minimize memory footprint/maximize LruCache)

Reply:
IMO, any hostname resolved by DNS relay should be cached, otherwise we cannot apply hostname rules to all of them.

Reply:
That I agree. These three lines are my attempt at an optimization (it should not change behavior at all).

Reply:
Okay, I think I can update the cache in `check_ip_in_proxy_list` to solve this.